test/unit/middlewares/egress-tracker.spec.js

```diff
@@ -0,0 +1,462 @@
+      expect(accountingRecordMethodStub.args[0][1], 'second argument should be the total bytes').to.equal(totalBytes)
+    })
+
+    it('should correctly track egress for responses with chunked transfer encoding', async () => {
+      const chunk1 = new TextEncoder().encode('Hello, ')
+      const chunk2 = new TextEncoder().encode('world!')
+      const totalBytes = Buffer.byteLength(chunk1) + Buffer.byteLength(chunk2)
+
+      const mockResponse = new Response(new ReadableStream({
+        start (controller) {
+          controller.enqueue(chunk1)
+          controller.enqueue(chunk2)
+          controller.close()
+        }
+      }), { status: 200 })
+
+      const innerHandler = sinon.stub().resolves(mockResponse)
+      const handler = withEgressHandler(innerHandler)
+      const request = await createRequest()
+
+      const response = await handler(request, env, ctx)
+      const responseBody = await response.text()
+
+      expect(response.status).to.equal(200)
+      expect(responseBody).to.equal('Hello, world!')
+      expect(accountingRecordMethodStub.calledOnce, 'record should be called once').to.be.true
+      expect(accountingRecordMethodStub.args[0][1], 'second argument should be the total bytes').to.equal(totalBytes)
+    })
+
+    it('should record egress bytes for a CAR file request', async () => {
+      // Simulate a CAR file content
+      const carContent = new Blob([randomBytes(256)])
+      const { shards } = await builder.add(carContent)
+      assert.equal(shards.length, 1)
+
+      const key = toBlobKey(shards[0].multihash)
+      /** @type {Uint8Array | undefined} */
+      const carBytes = await bucket.get(key)
+      expect(carBytes).to.be.not.undefined
+      expect(carBytes).to.be.instanceOf(Uint8Array)
+      const expectedTotalBytes = carBytes.byteLength
+
+      // Mock a response with the CAR file content
+      const mockResponse = new Response(new ReadableStream({
+        start (controller) {
+          controller.enqueue(carBytes)
+          controller.close()
+        }
+      }), {
+        status: 200,
+        headers: { 'Content-Type': 'application/vnd.ipld.car; version=1;' }
+      })
+
+      const innerHandler = sinon.stub().resolves(mockResponse)
+      const handler = withEgressHandler(innerHandler)
+      const request = await createRequest()
+
+      const response = await handler(request, env, ctx)
+      expect(response.status).to.equal(200)
+
+      // Consume the response body by reading the CAR file
+      const source = /** @type {ReadableStream<Uint8Array>} */ (await response.body)
+
+      /** @type {(import('carstream').Block & import('carstream').Position)[]} */
+      const blocks = []
+      await source
+        .pipeThrough(new CARReaderStream())
+        .pipeTo(new WritableStream({
+          write: (block) => { blocks.push(block) }
+        }))
+
+      // expect(blocks[0].bytes).to.deep.equal(carBytes) - FIXME (fforbeck): how to get the correct byte count?
```

Comment:
Still need to figure out how to check the byte count of the returned CAR file.

Reply:
@Peeja / @hannahhoward, any advice on comparing the total bytes of the returned CAR file with `expectedTotalBytes`? I suspect I'm not converting the file correctly. Any tips would be helpful!